src/commands/test/test_scheduler.py

```diff
@@ -0,0 +1,49 @@
+import multiprocessing
+import signal
+from typing import TYPE_CHECKING, Callable, List
+
+from src.commands.test.test_results_queue import TestResultsQueue
+from src.commands.test.test_runner import TestRunner
+from src.commands.test.testing_live_logger import TestingLiveLogger
+
+if TYPE_CHECKING:
+    from src.commands.test.test_collector import TestCollector
+
+
+class TestScheduler:
+    def __init__(
+        self,
+        live_logger: TestingLiveLogger,
+        worker: Callable[
+            [TestRunner.WorkerArgs],
+            None,
+        ],
+    ):
+        self._live_logger = live_logger
+        self._worker = worker
+
+    def run(
+        self, test_collector_result: "TestCollector.Result", include_paths: List[str]
+    ):
+        with multiprocessing.Manager() as manager:
+            test_results_queue = TestResultsQueue(manager.Queue())
+            setups: List[TestRunner.WorkerArgs] = [
+                TestRunner.WorkerArgs(
+                    test_suite,
+                    test_results_queue,
+                    include_paths,
+                )
+                for test_suite in test_collector_result.test_suites
+            ]
+
+            try:
+                with multiprocessing.Pool(
+                    multiprocessing.cpu_count(),
+                    lambda: signal.signal(
+                        signal.SIGINT, signal.SIG_IGN
+                    ),  # prevents showing a stacktrace on cmd/ctrl + c
+                ) as pool:
+                    pool.map_async(self._worker, setups)
+                    self._live_logger.log(test_results_queue, test_collector_result)
```

Comment:
> If the process throws an unexpected error, Protostar should crash. It should be a rare case, therefore we shouldn't care about logging results.

You probably misunderstood me here, if you won't provide get protostar **won't** crash in case of unexpected error. 

Please add it.

Reply:
I'll leave this `get` just to avoid changing the code.